cmd/legacy_param_converter_test.go

```diff
@@ -0,0 +1,323 @@
+				Logging: cfg.LoggingConfig{
+					Severity: "INFO",
+					Format:   "text",
+					FilePath: cfg.ResolvedPath(path.Join(os.Getenv("HOME"), "Documents/log-file.txt")),
+					LogRotate: cfg.LogRotateLoggingConfig{
+						MaxFileSizeMb:   20,
+						BackupFileCount: 2,
+						Compress:        true,
+					},
+				},
+				FileCache: cfg.FileCacheConfig{
+					MaxSizeMb:                  20,
+					CacheFileForRangeRead:      true,
+					EnableParallelDownloads:    true,
+					DownloadParallelismPerFile: 3,
+					MaxDownloadParallelism:     6,
+					ReadRequestSizeMb:          9,
+					EnableCrcCheck:             true,
+				},
+				CacheDir: cfg.ResolvedPath(path.Join(os.Getenv("HOME"), "cache-dir")),
+				MetadataCache: cfg.MetadataCacheConfig{
+					TtlSecs:            200,
+					TypeCacheMaxSizeMb: 7,
+					StatCacheMaxSizeMb: 4,
+				},
+				List: cfg.ListConfig{
+					EnableEmptyManagedFolders: true,
+					KernelListCacheTtlSecs:    30,
+				},
+				GcsConnection: cfg.GcsConnectionConfig{
+					GrpcConnPoolSize: 29,
+					ClientProtocol:   cfg.Protocol("grpc")},
+				GcsAuth:   cfg.GcsAuthConfig{AnonymousAccess: true},
+				EnableHns: true,
+				FileSystem: cfg.FileSystemConfig{
+					DisableParallelDirops: true,
+					IgnoreInterrupts:      true,
+				},
+			},
+		},
+		{testName: "Test overlapping flags and configs set.",
+			legacyFlagStorage: &flagStorage{
+				LogFile:                   "~/Documents/log-flag.txt",
+				LogFormat:                 "json",
+				IgnoreInterrupts:          false,
+				AnonymousAccess:           false,
+				KernelListCacheTtlSeconds: -1,
+				ClientProtocol:            mountpkg.HTTP2,
+			},
+			isFlagSet: map[string]bool{
+				"log-file":                   true,
+				"log-format":                 true,
+				"ignore-interrupts":          true,
+				"anonymous-access":           true,
+				"kernel-list-cache-ttl-secs": true,
+			},
+			legacyMountConfig: &config.MountConfig{
+				LogConfig: config.LogConfig{
+					FilePath: "~/Documents/log-config.txt",
+					Format:   "text",
+					Severity: "INFO",
+				},
+				FileSystemConfig: config.FileSystemConfig{
+					IgnoreInterrupts: true,
+				},
+				GCSAuth: config.GCSAuth{
+					AnonymousAccess: true,
+				},
+				ListConfig: config.ListConfig{
+					KernelListCacheTtlSeconds: 100,
+				},
+			},
+			expectedConfig: &cfg.Config{
+				Logging: cfg.LoggingConfig{
+					FilePath: cfg.ResolvedPath(path.Join(os.Getenv("HOME"), "/Documents/log-flag.txt")),
+					Format:   "json",
+					Severity: "INFO",
+				},
+				FileSystem: cfg.FileSystemConfig{
+					IgnoreInterrupts: false,
+				},
+				GcsAuth: cfg.GcsAuthConfig{
+					AnonymousAccess: false,
+				},
+				List: cfg.ListConfig{
+					KernelListCacheTtlSecs: -1,
+				},
+				GcsConnection: cfg.GcsConnectionConfig{
+					ClientProtocol: cfg.Protocol("http2"),
+				},
+			},
+		},
+	}
+
+	for _, tt := range populateConfigFromLegacyFlags {
+		t.Run(tt.testName, func(m *testing.T) {
```

Comment:
You can shadow the variable here.
```suggestion
		t.Run(tt.testName, func(t *testing.T) {
```